lucene/core/src/java/org/apache/lucene/util/bkd/MutablePointsReaderUtils.java

```diff
@@ -39,13 +41,23 @@
   public static void sort(
       BKDConfig config, int maxDoc, MutablePointValues reader, int from, int to) {
     final int bitsPerDocId = PackedInts.bitsRequired(maxDoc - 1);
```

Comment:
I wonder if we could check whether doc IDs are sorted right above this line, and then adapt the number of compared bytes? Then we should be able to get rid of `isEnableStableSort` and `doClosureIfStableSortEnabled`?

```suggestion
    // In the common case that index sorting is not enabled, doc IDs come in order, which enables us to check fewer bytes for sorting
    boolean sortedDocIDs = true;
    for (int i = from + 1; i < to; i++) {
      if (reader.getDocID(i - 1) > reader.getDocID(i)) {
        sortedDocIDs = false;
        break;
      }
    }
    final int bitsPerDocId = sorted ? 0 : PackedInts.bitsRequired(maxDoc - 1);
```

Reply:
Is this a little bit better?
```
bool canUseStableSort = check....
if (canUseStableSort) {
  StableMsbRadixSort.sort(..)
} else {
  MsbRadixSort.sort(..)
}
```

Reply:
Good question, my gut feeling is that the benefits of switching between the stable and the unstable radix sort are not worth the code complexity. Let's always use the stable sort?

Reply:
Will refine after we decide whether to always use stable sort.